azurerm/internal/services/datafactory/resource_arm_data_factory_dataset_azure_blob.go

```diff
@@ -0,0 +1,348 @@
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"folder": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"additional_properties": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"schema_column": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"Byte",
+								"Byte[]",
+								"Boolean",
+								"Date",
+								"DateTime",
+								"DateTimeOffset",
+								"Decimal",
+								"Double",
+								"Guid",
+								"Int16",
+								"Int32",
+								"Int64",
+								"Single",
+								"String",
+								"TimeSpan",
+							}, false),
+						},
+						"description": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceArmDataFactoryDatasetAzureBlobCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.DatasetClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	dataFactoryName := d.Get("data_factory_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Data Factory Dataset Azure Blob %q (Data Factory %q / Resource Group %q): %s", name, dataFactoryName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_factory_dataset_delimited_text", *existing.ID)
+		}
+	}
+
+	azureBlobDatasetProperties := datafactory.AzureBlobDatasetTypeProperties{
+		FolderPath: d.Get("path").(string),
+		FileName:   d.Get("filename").(string),
+	}
+
+	linkedServiceName := d.Get("linked_service_name").(string)
+	linkedServiceType := "LinkedServiceReference"
+	linkedService := &datafactory.LinkedServiceReference{
+		ReferenceName: &linkedServiceName,
+		Type:          &linkedServiceType,
+	}
+
+	description := d.Get("description").(string)
+	// TODO
+	azureBlobTableset := datafactory.AzureBlobDataset{
+		AzureBlobDatasetTypeProperties: &azureBlobDatasetProperties,
+		LinkedServiceName:              linkedService,
+		Description:                    &description,
+	}
```

Comment:
As mentioned before for consistency with other resources could we combine all this into 1 assignment?
```suggestion

	azureBlobTableset := datafactory.AzureBlobDataset{
		AzureBlobDatasetTypeProperties: &datafactory.AzureBlobDatasetTypeProperties{
			FolderPath: d.Get("path").(string),
			FileName:   d.Get("filename").(string),
		},
		LinkedServiceName: &datafactory.LinkedServiceReference{
			ReferenceName: utils.Strings(d.Get("linked_service_name").(string)),
			Type:          utils.String("LinkedServiceReference"),
		},
		Description:                    &utils.Streing(d.Get("description").(string)),
	}
```